typegate/tests/runtimes/graphql/graphql_test.ts

```diff
@@ -1,11 +1,211 @@
 // Copyright Metatype OÜ, licensed under the Elastic License 2.0.
 // SPDX-License-Identifier: Elastic-2.0
 
-import { Meta } from "../../utils/mod.ts";
+import { QueryEngine } from "../../../src/engine/query_engine.ts";
+import { randomPGConnStr } from "../../utils/database.ts";
+import { dropSchemas, recreateMigrations } from "../../utils/migrations.ts";
+import { gql, Meta } from "../../utils/mod.ts";
+
+const PYTHON_TG_PATH = "runtimes/graphql/typegraphs/python/graphql.py";
+const TS_TG_PATH = "runtimes/graphql/typegraphs/deno/graphql.ts";
 
 Meta.test("Typegraph generation with GraphQL runtime", async (t) => {
   await t.assertSameTypegraphs(
-    "runtimes/graphql/typegraphs/deno/graphql.ts",
-    "runtimes/graphql/typegraphs/python/graphql.py",
+    TS_TG_PATH,
+    PYTHON_TG_PATH,
   );
 });
+
+async function testEngine(engine: QueryEngine) {
+  // get users
+  await gql`
+    {
+      users {
+        data {
+          id
+          name
+        }
+      }
+    }
+  `
+    .expectData({
+      users: {
+        data: [
+          {
+            id: "1",
+            name: "Leanne Graham",
+          },
+          {
+            id: "2",
+            name: "Ervin Howell",
+          },
+          {
+            id: "3",
+            name: "Clementine Bauch",
+          },
+          {
+            id: "4",
+            name: "Patricia Lebsack",
+          },
+          {
+            id: "5",
+            name: "Chelsey Dietrich",
+          },
+          {
+            id: "6",
+            name: "Mrs. Dennis Schulist",
+          },
+          {
+            id: "7",
+            name: "Kurtis Weissnat",
+          },
+          {
+            id: "8",
+            name: "Nicholas Runolfsdottir V",
+          },
+          {
+            id: "9",
+            name: "Glenna Reichert",
+          },
+          {
+            id: "10",
+            name: "Clementina DuBuque",
+          },
+        ],
+      },
+    })
+    .on(engine);
+
+  // get user
+  await gql`
+    {
+      user(id: "1") {
+        id
+        name
+      }
+    }
+  `
+    .expectData({
+      user: {
+        id: "1",
+        name: "Leanne Graham",
+      },
+    })
+    .on(engine);
+
+  // Mutation
+  await gql`
+    mutation {
+      create_message(data: { title: "Hey", user_id: "1" }) {
+        id
+        title
+        user_id
+      }
+    }
+  `
+    .expectData({
+      create_message: {
+        id: 1,
+        title: "Hey",
+        user_id: "1",
+      },
+    })
+    .on(engine);
+
+  // TODO: Fails because metatype type system is not recognized by outside APIs
```

Comment:
not sure to understand exactly what it means?

Reply:
to elaborate more, 

<img width="741" alt="Screenshot 2024-06-20 at 10 39 47 AM" src="https://github.com/metatypedev/metatype/assets/77430541/202ccffb-3078-4aa2-8793-f6a730c33f0f">

but found out that the type should be wrapped looking into `typegate/tests/graphql/graphql_test.ts`, applying that now.